hedera-node/hedera-mono-service/src/main/java/com/hedera/services/ledger/accounts/AliasManager.java

```diff
@@ -240,7 +244,30 @@ public void forgetEvmAddress(final ByteString alias) {
      * @return EntityNum mapped to the given alias.
      */
     public EntityNum lookupIdBy(final ByteString alias) {
-        return curAliases().getOrDefault(alias, MISSING_NUM);
+        var entityNum = curAliases().getOrDefault(alias, MISSING_NUM);
+        if (alias.size() > EVM_ADDRESS_LEN && entityNum == MISSING_NUM) {
+            // if we don't find entity num for key alias we can try to derive EVM address from it
+            // and look it up
+            try {
```

Comment:
This duplicates the logic starting at L221 above; we should extract that into a helper function and use it here.

Reply:
Extracted duplicated logic to `keyAliasToEVMAddress`.